src/odemis/util/img.py

```diff
@@ -1201,3 +1201,41 @@ def assembleZCube(images, zlevels):
         ret = DataArray(ret, metadata3d)
 
         return ret
+
+
+def apply_flood_fill(array, start):
+    """
+    Flood fills a 2-Dimensional numpy array with truth values from a given start position with the 4-connected method.
+    The input array contains a shape represented with truth values for the borders of the shape.
+
+    :param array: (ndarray(bool)) binary array of size mxn containing the shape to be filled
+    :param start: (tuple(int, int)) position from which to start flood fill
+    :return: (ndarray(bool)) array of size mxn containing the filled shape, with filled values True
+    """
+    if start >= array.shape:
+        raise ValueError(f"Start position of {start} does not lie within the array.")
+
+    array = array.copy()
```

Comment:
`array` is a python standard module, I think it's good to be a bit careful with naming things the same as other libraries. I would suggest to rename this, maybe to something like `input_array` or `shape_array`